src/cbc_sdk/workload/nsx_remediation.py

```diff
@@ -0,0 +1,139 @@
+            real_device_ids = device_ids
+        elif isinstance(device_ids, int):
+            real_device_ids = [device_ids]
+        else:
+            raise ApiError("device_ids of invalid type")
+        if tag not in NSXRemediationJob.VALID_TAGS:
+            raise ApiError(f"invalid tag: {tag}")
+
+        request_body = {'device_ids': real_device_ids, 'action_type': 'NSX_REMEDIATION',
+                        'options': {'toggle': 'ON' if set_tag else 'OFF', 'tag': tag}}
+        url = "/applianceservice/v1/orgs/{}/device_actions".format(cb.credentials.org_key)
+        response = cb.post_object(url, body=request_body)
+        if response.status_code != 201:
+            raise ServerError(response.status_code,
+                              f"could not start remediation request, error code {response.status_code}")
+        job_ids = response.json().get('job_ids', [])
+        if job_ids:
+            return NSXRemediationJob(cb, job_ids)
+        raise NSXJobError("No jobs started")
+
+    def _poll_status(self):
+        """
+        Polls the current status of all running jobs, saves it, and eliminates all jobs that are no longer running.
+
+        Required Permissions:
+            appliances.registration(READ)
+
+        Returns:
+            bool: True if at least one job is still running, otherwise False.
+        """
+        done_jobs = []
+        for job in self._running_jobs:
+            url = "/applianceservice/v1/orgs/{}/jobs/{}/status".format(self._cb.credentials.org_key, job)
+            response = self._cb.get_object(url)
+            self._status[job] = response
+            if response['status'] not in NSXRemediationJob.RUNNING_STATUSES:
+                done_jobs.append(job)
+
+        for job in done_jobs:
+            self._running_jobs.remove(job)
+        return len(self._running_jobs) > 0
+
+    def await_result(self):
+        """
+        Waits for all running jobs to be completed and returns the final status.
+
+        Required Permissions:
+            appliances.registration(READ)
+
+        Returns:
+            dict: The final status, mapping individual job IDs to status value dicts.
+        """
+        while self._poll_status():
+            time.sleep(1)
+        return self._status
+
+    def async_await_result(self):
+        """
+        Sets up a Future which can be used to wait asynchronously for all running jobs to be completed.
+
+        Required Permissions:
+            appliances.registration(READ)
+
+        Returns:
+            Future: A future representing the job and its results.
+        """
+        return self._cb._async_submit(lambda arg, kwarg: arg[0].await_result(), self)
+
+    @property
+    def status(self):
+        """
+        Returns the current status.
+
+        Returns:
+            dict: The current status, mapping individual job IDs to status value dicts.
+        """
+        return self._status
```

Comment:
If the user calls this before using either of the `await_result` functions then the _status will be all None. Should we call `_poll_status` once before returning. We could even check if _running_jobs > 0 such that we don't need to poll if it's already finished

Reply:
We call `_poll_status` at the end of the `__init__` method, so by the time the object is constructed, it's already been polled once.